src/server/scripts/Outland/HellfireCitadel/MagtheridonsLair/boss_magtheridon.cpp

```diff
@@ -16,617 +15,590 @@
+        void UpdateAI(uint32 diff) override
         {
-            npc_hellfire_channelerAI(Creature* creature) : ScriptedAI(creature)
-            {
-                Initialize();
-                instance = creature->GetInstanceScript();
-            }
+            _scheduler.Update(diff);
+        }
 
-            void Initialize()
-            {
-                ShadowBoltVolley_Timer = urand(8000, 10000);
-                DarkMending_Timer = 10000;
-                Fear_Timer = urand(15000, 20000);
-                Infernal_Timer = urand(10000, 50000);
+    private:
+        TaskScheduler _scheduler;
+    };
 
-                Check_Timer = 5000;
-            }
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_magtheridon_roomAI(creature);
+    }
+};
 
-            InstanceScript* instance;
+class go_manticron_cube : public GameObjectScript
+{
+public:
+    go_manticron_cube() : GameObjectScript("go_manticron_cube") { }
 
-            uint32 ShadowBoltVolley_Timer;
-            uint32 DarkMending_Timer;
-            uint32 Fear_Timer;
-            uint32 Infernal_Timer;
+    bool OnGossipHello(Player* player, GameObject* /*go*/) override
+    {
+        if (player->HasAura(SPELL_MIND_EXHAUSTION) || player->HasAura(SPELL_SHADOW_GRASP))
+            return true;
 
-            uint32 Check_Timer;
+        if (Creature* trigger = player->FindNearestCreature(NPC_HELFIRE_RAID_TRIGGER, 10.0f))
+            trigger->CastSpell((Unit*)nullptr, SPELL_SHADOW_GRASP_VISUAL);
 
-            void Reset() override
-            {
-                Initialize();
-            }
+        player->CastSpell((Unit*)nullptr, SPELL_SHADOW_GRASP, true);
+        return true;
+    }
+};
 
-            void EnterCombat(Unit* /*who*/) override
-            {
-                instance->SetData(DATA_CHANNELER_EVENT, IN_PROGRESS);
+class spell_magtheridon_blaze_target : public SpellScriptLoader
+{
+    public:
+        spell_magtheridon_blaze_target() : SpellScriptLoader("spell_magtheridon_blaze_target") { }
 
-                me->InterruptNonMeleeSpells(false);
-                DoZoneInCombat();
-            }
+        class spell_magtheridon_blaze_target_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_magtheridon_blaze_target_SpellScript);
 
-            void JustReachedHome() override
+            bool Validate(SpellInfo const* /*spell*/) override
             {
-                instance->SetData(DATA_CHANNELER_EVENT, NOT_STARTED);
-
-                DoCast(me, SPELL_SHADOW_GRASP_C, false);
+                if (!sSpellMgr->GetSpellInfo(SPELL_BLAZE))
+                    return false;
+                return true;
             }
 
-            void JustSummoned(Creature* summon) override
+            void HandleAura()
             {
-                summon->AI()->AttackStart(me->GetVictim());
+                GetHitUnit()->CastSpell(GetHitUnit(), SPELL_BLAZE);
```

Comment:
Thats what we do but no clue what it actually does in the client (and as you know it used to cause client crashes for spells cast by players)